daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Pretty.scala

```diff
@@ -78,14 +62,14 @@ private[lf] object Pretty {
         text("Update failed due to fetch of an inactive contract") & prettyContractId(coid) &
           char('(') + (prettyTypeConName(tid)) + text(").") /
           text(s"The contract had been consumed in sub-transaction #$consumedBy:") +
-          (ptx.nodes.get(consumedBy) match {
+          (consumedBy match {
             // FIXME(JM): How should we show this? If the node pointed to by consumedBy
             // is not in nodes, it must not be done yet, hence the contract is recursively
             // exercised.
```

Comment:
> FIXME(JM): How should we show this?
When an exercise is opened but not closed.
> If the node pointed to by consumedBy is not in nodes, it must not be done yet, hence the contract is recursively exercised.

No, if the body of an exercise fails, the Node Id of the the exercise was not added to `ptx.node`.
On the other hand a "recursive exercised" can never happen, because the contract is consumed at the beginning of an exercise before its body is evaluated.